admin-console/src/main/webapp/src/components/CreateEditModal.tsx

```diff
@@ -0,0 +1,87 @@
+import { ActionGroup, Button, Form, FormGroup, HelperText, HelperTextItem, Modal, ModalVariant,
+    Spinner, TextArea, TextInput } from '@patternfly/react-core';
+import React from 'react';
+
+import { useCreateEventType } from '../services/CreateEventTypes';
+import { EventType } from '../types/Notifications';
+
+interface CreateEditModalProps {
+    isEdit: boolean;
+    showModal: boolean;
+    applicationName?: string;
+    initialEventType?: EventType;
+    isLoading?: boolean;
+    onClose: () => void;
+    onSubmit: (eventType: EventType) => void;
+    eventTypeQuery: unknown;
+
+}
+
+export const CreateEditModal: React.FunctionComponent<CreateEditModalProps> = (props) => {
+
+    const [ eventType, setEventType ] = React.useState<Partial<EventType>>(props.initialEventType ?? {});
+    const newEvent = useCreateEventType();
+
+    const handleChange = (value: string, event: React.FormEvent<HTMLInputElement> | React.FormEvent<HTMLTextAreaElement>) => {
+        const target = event.target as HTMLInputElement;
+        setEventType(prev => ({ ...prev, [target.name]: target.value }));
+    };
+
+    const onSubmitLocal = React.useCallback(() => {
+        props.onSubmit(eventType); }, [ eventType, props ]);
+
+    React.useEffect(() => {
+        setEventType(props.initialEventType);
```

Comment:
I am getting an error here `Argument of type 'Partial<EventType> | undefined' is not assignable to parameter of type 'SetStateAction<Partial<EventType>>'.` on the props.initialEventType

Reply:
I suppose it's complaining because it could be undefined. So you would have to provide a fallback value
```suggestion
        setEventType(props.initialEventType ?? {});
```